### PR TITLE
Improve telemetry instrumentation on core flows

### PR DESCRIPTION
**Pull request by wcalderipe**
This PR improves telemetry instrumentation on core flows as well as adds business metrics to answer a few questions:

- How many wallets per client and overall?
- How many sign<Message/Transaction/TypedData/Raw> per client and overall?
- How many authorization flows were initiated by a client and overall?
- How many authorization flows were permitted by a client and overall?

## Changelog

- Moved shared constants `OTEL_ATTR_CLIENT_ID`, `REQUEST_HEADER_CLIENT_ID`, and `REQUEST_HEADER_CLIENT_SECRET` to`nestjs-shared` package. 
- Added middleware in all applications to set the value from the request header `x-client-id` to the active span.
- Added granular spans on the evaluation flow of the policy engine.
- Rename the service name of the Armory from `auth` to `armory`.

### CODE REVIEW

File: apps/armory/src/managed-data-store/core/service/entity-data-store.service.ts [hunk 1]
```diff
@@ -27,10 +27,10 @@ export class EntityDataStoreService extends SignatureService {
   }
 
   async getEntities(clientId: string): Promise<EntityStore | null> {
```
Comment (Ptroger) [inline apps/armory/src/managed-data-store/core/service/entity-data-store.service.ts:29]: Looks easy to setup new ones and really easy to read aswell, nice

File: apps/armory/src/orchestration/core/service/authorization-request.service.ts [hunk 2]
```diff
@@ -48,10 +52,16 @@ export class AuthorizationRequestService {
     private priceService: PriceService,
     private clusterService: ClusterService,
     private feedService: FeedService,
-    private logger: LoggerService
-  ) {}
+    private logger: LoggerService,
+    @Inject(MetricService) private metricService: MetricService
+  ) {
+    this.createCounter = this.metricService.createCounter('authorization_request_create_count')
```
Comment (Ptroger) [inline apps/armory/src/orchestration/core/service/authorization-request.service.ts:58]: Maybe rename createCounter attribute to something different than metricService create counter method ? Not a big deal but the first few seconds I was confused

File: apps/policy-engine/README.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ Policy decision point for fine-grained authorization in web3.0.
 - [Open Policy Agent (OPA) binary version >=
   0.69](https://www.openpolicyagent.org/docs/latest/#1-download-opa) installed
   and accessible in your `$PATH`.
+- [Regal](https://github.com/StyraInc/regal?tab=readme-ov-file#getting-started)
+  for linting Rego code.
```
Comment (Ptroger) [inline apps/policy-engine/README.md:11]: I guess I should have put that in my latest Regal version PR, ty for adding that

**Review by Ptroger — APPROVED** [review]
That looks super nice. Seem easy to setup new metrics from there. Appreciated to take a look on how you tested these metric and trace services with the stateful ones. I left a comment about one naming and one question.
Ty for the code comments from earlier PR with links to documentation, it was helpful.

File: apps/armory/src/orchestration/core/service/authorization-request.service.ts (hunk 2 above)
Comment (wcalderipe) [reply apps/armory/src/orchestration/core/service/authorization-request.service.ts:58]: The class name carries the context needed to understand what "create" means. However, when the class has more than one responsibility, like in the KeyGenerationService, a more verbose name makes sense.

File: apps/armory/src/armory.constant.ts [hunk 4]
```diff
@@ -49,8 +55,6 @@ export const DEFAULT_HTTP_MODULE_PROVIDERS = [
 // Headers
 //
 
-export const REQUEST_HEADER_CLIENT_ID = 'x-client-id'
-export const REQUEST_HEADER_CLIENT_SECRET = 'x-client-secret'
 export const REQUEST_HEADER_API_KEY = 'x-api-key'
```
[the remark this answers is not in the document]
Comment (wcalderipe) [reply apps/armory/src/armory.constant.ts:58]: I felt that was specific to the Armory, so I left it there.

**Pull request merged**